src/python/pants/engine/rules.py

```diff
@@ -302,6 +302,7 @@ def get_some_union_type(x):
   assert isinstance(cls, type)
   return type(cls.__name__, (cls,), {
     '_is_union': True,
+    'union_description': cls.__doc__,
```

Comment:
How about a fallback to the `cls.__name__` if `cls.__doc__` is empty? The current erroring test shows off a fairly unhelpful case of an empty doc string.

Reply:
Good idea, added